pallets/runtime/common/src/bridge.rs

```diff
@@ -425,83 +426,110 @@ decl_module! {
+        #[weight = SimpleDispatchInfo::FixedOperational(250_000)]
         pub fn handle_bridge_tx(origin, bridge_tx: BridgeTx<T::AccountId, T::Balance>) ->
             DispatchResult
         {
             let sender = ensure_signed(origin)?;
-            let mut tx_details = Self::bridge_tx_details(&bridge_tx.recipient, &bridge_tx.nonce);
-            match tx_details.status {
-                // New bridge tx
-                BridgeTxStatus::Absent => {
-                    //TODO: Review admin permissions to handle bridge txs before mainnet
-                    ensure!(sender == Self::controller() || sender == Self::admin(), Error::<T>::BadCaller);
-                    let timelock = Self::timelock();
-                    if timelock.is_zero() {
-                        return Self::handle_bridge_tx_now(bridge_tx, false);
-                    } else {
-                        return Self::handle_bridge_tx_later(bridge_tx, timelock);
-                    }
-                }
-                // Pending cdd bridge tx
-                BridgeTxStatus::Pending(_) => {
-                    return Self::handle_bridge_tx_now(bridge_tx, true);
-                }
-                // Pre frozen tx. We just set the correct amount.
-                BridgeTxStatus::Frozen => {
-                    //TODO: Review admin permissions to handle bridge txs before mainnet
-                    ensure!(sender == Self::controller() || sender == Self::admin(), Error::<T>::BadCaller);
-                    tx_details.amount = bridge_tx.amount;
-                    <BridgeTxDetails<T>>::insert(&bridge_tx.recipient, &bridge_tx.nonce, tx_details);
-                    Ok(())
-                }
-                BridgeTxStatus::Timelocked => {
-                    return Err(Error::<T>::TimelockedTx.into());
-                }
-                BridgeTxStatus::Handled => {
-                    return Err(Error::<T>::ProposalAlreadyHandled.into());
-                }
+            Self::handle_signed_bridge_tx(&sender, bridge_tx)
+        }
+
+        /// Handles a vector of approved bridge transaction proposals. The vector is processed until
+        /// the first proposal which causes an error, in which case the error is returned and the
+        /// rest of proposals are not processed.
+        ///
+        /// # Weight
+        /// `50_000 + 200_000 * bridge_txs.len()`
+        #[weight = FunctionOf(
+            |(bridge_txs,): (
+                &Vec<BridgeTx<T::AccountId, T::Balance>>,
+            )| {
+                50_000 + 200_000 * u32::try_from(bridge_txs.len()).unwrap_or_default()
+            },
+            DispatchClass::Operational,
+            true
+        )]
+        pub fn handle_bridge_txs(origin, bridge_txs: Vec<BridgeTx<T::AccountId, T::Balance>>) ->
+            DispatchResult
+        {
+            let sender = ensure_signed(origin)?;
+            for tx in bridge_txs {
+                Self::handle_signed_bridge_tx(&sender, tx)?;
```

Comment:
Ideally we would process all bridge_txs rather than failing on the first error. i.e. so if the first bridge_tx in the vector is "bad", we still process the others.

Reply:
We'll then need some (easy) way to get the result(success/failure) of all of the txs. Perhaps an event at the end?